### PR TITLE
Mapping uniqueid with cartography

### PR DESCRIPTION
**Pull request by 01manasrathi**

### CODE REVIEW

File: cartography/intel/bitbucket/members.py [hunk 1]
```diff
@@ -6,41 +6,49 @@
...
+def load_members_data(session: neo4j.Session, members_data: List[Dict], common_job_parameters: Dict) -> None:
+    session.write_transaction(_load_members_data, members_data, common_job_parameters)
 
 
-def _load_members_data(tx: neo4j.Transaction,members_data:List[Dict],common_job_parameters:Dict):
-    ingest_workspace="""
+def _load_members_data(tx: neo4j.Transaction, members_data: List[Dict], common_job_parameters: Dict):
+    ingest_workspace = """
     UNWIND $membersData as member
     MERGE (mem:BitbucketMember{id: member.user.uuid})
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/members.py:53]: Instead of uuid use member.id as id here

File: cartography/intel/bitbucket/members.py [hunk 2]
```diff
@@ -83,7 +91,7 @@ def sync(
     :return: Nothing
     """
     logger.info("Syncing Bitbucket All workspace members")
-    workspace_members=get_workspace_members(bitbucket_access_token,workspace_name)
-    workspace_members=transform_members(workspace_members)
-    load_members_data(neo4j_session,workspace_members,common_job_parameters)
-    cleanup(neo4j_session,common_job_parameters)
+    workspace_members = get_workspace_members(bitbucket_access_token, workspace_name)
+    workspace_members = transform_members(workspace_members)
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/members.py:95]: workspace parameter missing here

File: cartography/intel/bitbucket/repositories.py [hunk 3]
```diff
@@ -12,40 +12,51 @@
...
-        repo['workspace']['uuid'] = repo['workspace']['uuid'].replace('{','').replace('}','')
-        repo['project']['uuid'] = repo['project']['uuid'].replace('{','').replace('}','')
-        repo['uuid'] = repo['uuid'].replace('{','').replace('}','')
+        repo['workspace']['uuid'] = repo['workspace']['uuid'].replace('{', '').replace('}', '')
+        repo['project']['uuid'] = repo['project']['uuid'].replace('{', '').replace('}', '')
+        repo['uuid'] = repo['uuid'].replace('{', '').replace('}', '')
+
+        data = {
+            "workspace": workspace,
+            "project": repo['project']['uuid'],
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/repositories.py:41]: Instead of uuid pass project name

File: cartography/intel/bitbucket/repositories.py [hunk 4]
```diff
@@ -12,40 +12,51 @@
...
+            "repository": repo['uuid']
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/repositories.py:42]: pass repository name instead of uuid

File: cartography/intel/bitbucket/repositories.py [hunk 5]
```diff
@@ -12,40 +12,51 @@
...
+
+        }
 
         if repo is not None and repo.get('mainbranch') is not None:
-            repo['default_branch'] = repo.get('mainbranch',{}).get('name',None)
+            repo['default_branch'] = repo.get('mainbranch', {}).get('name', None)
+
+        repo['id'] = bitbucket_linker.get_unique_id(service="bitbucket", data=data, resource_type="member")
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/repositories.py:49]: Check `resource_type` parameter value. we should pass `repository`

File: cartography/intel/bitbucket/repositories.py [hunk 6]
```diff
@@ -12,40 +12,51 @@
...
+def load_repositories_data(session: neo4j.Session, repos_data: List[Dict], common_job_parameters: Dict) -> None:
+    session.write_transaction(_load_repositories_data, repos_data, common_job_parameters)
 
 
-def _load_repositories_data(tx: neo4j.Transaction,repos_data:List[Dict],common_job_parameters:Dict):
-    ingest_repositories="""
+def _load_repositories_data(tx: neo4j.Transaction, repos_data: List[Dict], common_job_parameters: Dict):
+    ingest_repositories = """
     UNWIND $reposData as repo
     MERGE (re:BitbucketRepository{id:repo.uuid})
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/repositories.py:61]: Instead of uuid use repo.id as id

Comment (01manasrathi) [reply cartography/intel/bitbucket/repositories.py:61]: ok

File: cartography/intel/bitbucket/workspace.py [hunk 7]
```diff
@@ -12,27 +12,34 @@
...
     return workspaces
 
 
 def transform_workspaces(workspaces: List[Dict]) -> List[Dict]:
     for workspace in workspaces:
-        workspace['uuid'] = workspace['uuid'].replace('{','').replace('}','')
+        workspace['uuid'] = workspace['uuid'].replace('{', '').replace('}', '')
 
+        data = {
+            workspace: "workspace"
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/workspace.py:37]: pass name of the `workspace` here

File: cartography/intel/bitbucket/workspace.py [hunk 8]
```diff
@@ -90,5 +97,5 @@ def sync(
     """
     logger.info("Syncing Bitbucket All workspaces")
 
-    workspaces=transform_workspaces(workspaces)
+    workspaces = transform_workspaces(workspaces)
     load_workspace_data(neo4j_session, workspaces, common_job_parameters)
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/workspace.py:101]: also make change in load query according id field

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
Minor changes needed

File: cartography/intel/bitbucket/members.py [hunk 9]
```diff
@@ -6,43 +6,51 @@
...
+    session.write_transaction(_load_members_data, members_data, common_job_parameters)
 
 
-def _load_members_data(tx: neo4j.Transaction,members_data:List[Dict],common_job_parameters:Dict):
-    ingest_workspace="""
+def _load_members_data(tx: neo4j.Transaction, members_data: List[Dict], common_job_parameters: Dict):
+    ingest_workspace = """
     UNWIND $membersData as member
-    MERGE (mem:BitbucketMember{id: member.user.uuid})
+    MERGE (mem:BitbucketMember{id: member.user.id})
```
Comment (ganeshjagdale9545) [inline cartography/intel/bitbucket/members.py:53]: `id` is a member's object

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
Minor changes needed 
also add clouduniqueid in `requirements.txt` file

File: docs/requirements.txt [hunk 10]
```diff
@@ -1,3 +1,4 @@
 sphinx==4.3.0
 sphinx-material==0.0.35
 m2r2==0.3.2
+clouduniqueid
```
Comment (ganeshjagdale9545) [inline docs/requirements.txt:4]: can you please add latest version number here?
you can get it from gihub repo latest release or from pypi.org

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
one minor change needed

**Review by ganeshjagdale9545 — APPROVED** [review]
Looks good

**Review by mpurusottamc — APPROVED** [review]
lgtm. merging for deployment.

**Pull request merged**